### PR TITLE
framework/st_things: Json modify and Fix memory leaks. 

### PR DESCRIPTION
**Pull request by KimDongEon**
1. framework/st_things: Json replace manufacturerid with manufacturerName
We are going to change sampleDevice.json.
So change the source code that gets the content from the json that changes.
The source code path to change is framework / src / st_things / things_stack / src / common / framework / things_data_manager.c.

2. framework/st_things: fix memory overload
In the second commit, there is over-allocation of memory. Modify this.

3. framework/st_things: MNID, SID Length-Limited Bug Fix
manufacture_name (MNID: 4 bytes) and setup_id (SID: 3 bytes) must be fixed sizes. If not the specified byte size, add logic to stop the operation.

4. apps/examples: Modify ST_Things Json File
Apply changes to Json file

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c [hunk 1]
```diff
@@ -1182,20 +1182,19 @@ static int parse_things_info_json(const char *filename)
...
 
-							if (NULL != manufature_id && NULL != setup_id) {
+							if (NULL != setup_id) {
 								is_artik = false;
 								if (artik->type == cJSON_True) {
 									is_artik = true;
 								}
 
-								THINGS_LOG_D(THINGS_INFO, TAG, "[configuration] manufature_id : %s / setup_id : %s / artik : %d", manufature_id->valuestring, setup_id->valuestring, is_artik);
+								THINGS_LOG_D(THINGS_INFO, TAG, "[configuration] manufature_name : %s / setup_id : %s / artik : %d", node->manufacturer_name, setup_id->valuestring, is_artik);
```
Comment (jsdosa) [inline framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: is that correct?
g_manufacturer_name = (char *)things_malloc(sizeof(char) * strlen(manufature_id->valuestring) + 1);
please, check the malloc lines~

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c (hunk 1 above)
Comment (KimDongEon) [reply framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: Memory is overload. I fix it.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by JeongJunSik** [comment]
Please check login of manufacture_name(MNID : 4 byte), setup_id(SID : 3 byte) length
This filed is fixed size

In Json parsing process, you need to check it is correct or invild data.

and also you need to check Device Name is correct or invilid.
becuase it is using when making softap ssid.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c [hunk 2]
```diff
@@ -1186,6 +1189,9 @@ static int parse_things_info_json(const char *filename)
 							cJSON *artik = cJSON_GetObjectItem(softap, KEY_CONFIGURATION_EASYSETUP_CONNECTIVITY_SOFTAP_ARTIK);
 
 							if (NULL != setup_id) {
+								if(strlen(setup_id->valuestring) != 3) {
+									return 1;
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: i think you shoud "return 0"

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c [hunk 3]
```diff
@@ -872,6 +872,9 @@ static int parse_things_info_json(const char *filename)
 					cJSON *vendor_id = cJSON_GetObjectItem(spec_platform, KEY_DEVICE_SPECIFICATION_PLATFORM_VENDORID);
 
 					if (NULL != manufacturer_name) {
+						if(strlen(manufacturer_name->valuestring) != 4) {
+							return 1;
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: i think you shoud "return 0"

Comment (JeongJunSik) [reply framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: also, please add a log the reason for return.

Comment (KimDongEon) [reply framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: modify

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c (hunk 2 above)
Comment (KimDongEon) [reply framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: modify

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c [hunk 4]
```diff
@@ -872,6 +872,10 @@ static int parse_things_info_json(const char *filename)
 					cJSON *vendor_id = cJSON_GetObjectItem(spec_platform, KEY_DEVICE_SPECIFICATION_PLATFORM_VENDORID);
 
 					if (NULL != manufacturer_name) {
+						if(strlen(manufacturer_name->valuestring) != 4) {
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: check coding rule

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c [hunk 5]
```diff
@@ -1186,6 +1190,10 @@ static int parse_things_info_json(const char *filename)
 							cJSON *artik = cJSON_GetObjectItem(softap, KEY_CONFIGURATION_EASYSETUP_CONNECTIVITY_SOFTAP_ARTIK);
 
 							if (NULL != setup_id) {
+								if(strlen(setup_id->valuestring) != 3) {
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: same as previous

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c (hunk 4 above)
Comment (KimDongEon) [reply framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: modify

File: framework/src/st_things/things_stack/src/common/framework/things_data_manager.c (hunk 5 above)
Comment (KimDongEon) [reply framework/src/st_things/things_stack/src/common/framework/things_data_manager.c]: modify

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**